storage/replica.go

```diff
@@ -3376,6 +3377,59 @@ func (r *Replica) maybeGossipSystemConfig() {
 	r.systemDBHash = hash
 }
 
+// maybeGossipNodeLiveness gossips information for all node liveness
+// records stored on this range. To scan and gossip, this replica
+// must hold the lease to a range which contains some or all of the
+// node liveness records. After scanning the records, it checks
+// against what's already in gossip and only gossips records which
+// are out of date.
+func (r *Replica) maybeGossipNodeLiveness(span roachpb.Span) {
+	if r.store.Gossip() == nil || !r.IsInitialized() {
+		return
+	}
+
+	if !r.ContainsKeyRange(span.Key, span.EndKey) {
+		return
+	}
+
+	if lease, _ := r.getLease(); !lease.OwnedBy(r.store.StoreID()) || !lease.Covers(r.store.Clock().Now()) {
+		// Do not gossip when a range lease is not held.
+		return
+	}
+
+	ba := roachpb.BatchRequest{}
+	ba.Timestamp = r.store.Clock().Now()
+	ba.Add(&roachpb.ScanRequest{Span: span})
+	br, trigger, pErr :=
+		r.executeBatch(r.ctx, storagebase.CmdIDKey(""), r.store.Engine(), nil, ba)
```

Comment:
This is the pattern used in gossiping the system span and @tschottdorf encouraged that I continue following it.